releasenotes/notes/42212.yaml

```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: traffic-management
+issue:
+  - 42212
+
+releaseNotes:
+- |
+  **Fixed** ServiceEntries with other revision labels were being processed.
```

Comment:
not only SE, but also any other CRs

Reply:
I was unsure how to word this, the only bug I saw was that SEs in other revisions were being processed fully. Processed is perhaps not the best word here.

I didn't try every other resource, but for the ones I did they were I guess "partially" processed, what would happen is that the event would come through from the crd client. Then they would not use the resource from the event directly, instead rebuild from scratch using the cache. The `List` and `Get` functions in the cache did have revision checks so places that used them would end up doing extra work, but not create incorrect configuration.